vue/components/ui/molecules/calendar/calendar.vue

```diff
@@ -0,0 +1,296 @@
+    background-color: #ecf0f3;
+}
+
+.calendar > .calendar-content .calendar-table .table-body .row .cell:not(.clickable) .circle {
+    color: #bac2cb;
+}
+
+.calendar > .calendar-content .calendar-table .table-body .row .cell.selected .circle {
+    background-color: #1d2631;
+    border: 3px solid #ecf0f3;
+    color: #ffffff;
+}
+
+.calendar .select ::v-deep .select-container .select-button {
+    background-color: #ffffff;
+}
+</style>
+
+<script>
+export const Calendar = {
+    props: {
+        /**
+         * The value of input date in String or Date format.
+         * Example: "2020/12/31"
+         */
+        value: {
+            type: Date | String,
+            default: null
+        },
+        /**
+         * Month labels to be used directly starting from January.
+         */
+        monthLabels: {
+            type: Array,
+            default: () => [
+                "January",
+                "February",
+                "March",
+                "April",
+                "May",
+                "June",
+                "July",
+                "August",
+                "September",
+                "October",
+                "November",
+                "December"
+            ]
+        },
+        /**
+         * Week labels to be used directly starting from Sunday.
+         */
+        weekLabels: {
+            type: Array,
+            default: () => ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"]
+        }
+    },
+    data: function() {
+        return {
+            valueData: this.value,
+            calendarVisibility: true,
+            day: 1,
+            month: new Date().getMonth(),
+            year: new Date().getFullYear()
+        };
+    },
+    computed: {
+        circleClass() {
+            return (day, month) => ({
+                clickable: this.month === month,
+                selected:
+                    this.selectedDay === day &&
+                    this.selectedMonth === month &&
+                    this.selectedYear === this.year
+            });
+        },
+        headerText() {
+            return `${this.monthName} ${this.year}`;
+        },
+        weekLabelsComputed() {
+            const weekLabels = [...this.weekLabels];
+            const lastDays = weekLabels.splice(0, 1);
+            weekLabels.push(...lastDays);
+            return weekLabels;
+        },
+        monthOptions() {
+            return this.monthLabels.map((month, index) => ({ value: index, label: month }));
+        },
+        monthName() {
+            return this.monthOptions[this.month]?.label;
+        },
+        weeks() {
+            const date = new Date(this.year, this.month, 1);
+            const firstMonday = date.getDay() === 0 ? -5 : date.getDate() - date.getDay() + 1;
+            date.setDate(firstMonday);
+            const weeks = new Array(6).fill(0).map(() =>
+                new Array(7).fill(0).map(day => {
+                    day = new Date(date);

```

Comment:
Avoid mutating variables, as it introduces cognitive load (= "wtf is 'day'" moment)

```suggestion
                new Array(7).fill(0).map(_ => {
                    const day = new Date(date);
```